src/V8JsiRuntime.cpp

```diff
@@ -1308,12 +1270,12 @@ std::string getFunctionName(v8::Isolate *isolate, v8::Local<v8::Function> func)
 
 jsi::Value
 V8Runtime::call(const jsi::Function &jsiFunc, const jsi::Value &jsThis, const jsi::Value *args, size_t count) {
-  _ISOLATE_CONTEXT_ENTER
+  IsolateLocker isolate_locker(this);

```

Comment:
>IsolateLocker isolate_locker(this); [](http://example.com/codeflow?start=2&length=35)

Do we use native promise in any scenarios ? How is the locker kept alive across multiple asynchronous micro tasks ?

Reply:
We don't use native promises in ReactNative (yet). Not sure what you mean about keeping the locker alive; we don't do that, we re-create the RAII wrapper on every entry into call (like we did for the HandleScope, Isolate Scope and Context Scope previously.